javalib/src/main/scala/java/io/CharArrayWriter.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.io
+
+import java.nio.CharBuffer
+
+class CharArrayWriter extends Writer {
+  private[this] var buf = CharBuffer.allocate(16)
```

Comment:
A `java.nio.CharBuffer` will not automatically grow when it runs out of space. Instead, it will throw an exception. That means currently you can't append more than 16 characters without throwing an exception. Given that, it's unclear to me that `CharBuffer` gives you anything that you can't have with a direct `Array[Char]`.

Perhaps you can take inspiration from [`ByteArrayOutputStream`'s implementation](https://github.com/scala-js/scala-js/blob/master/javalib/src/main/scala/java/io/ByteArrayOutputStream.scala), since `CharArrayWriter` essentially does the same but for the `Writer` interface.